matron/src/clocks/clock_link.c

```diff
@@ -63,6 +63,8 @@ static void *clock_link_run(void *p) {
                 clock_link_shared_data.requested_tempo = 0;
             }
 
+            ableton_link_enable(link, clock_link_shared_data.enabled);
```

Comment:
this is a [thread-safe method](https://github.com/Ableton/link/blob/master/include/ableton/Link.hpp#L98), so technically we don't need to pass the flag through clock_link_shared_data and just call ableton_link_enable with the appropriate parameter from clock_set_source.

Reply:
The nice thing about having the `ableton_link_enable()` call inside the mutex-guarded link thread is that we can guarantee that any pending tempo change requests are processed before we call enable.

Reply:
how does mutex ensures that? the only thread where these requests could come from is the single matron thread, so there can't be any race conditions around tempo.

Reply:
this is totally theoretical, just thinking out loud, but:
* change clock source from midi to link, param handler for clock_source runs
* handler calls `clock.link.set_tempo()`. link thread is sleeping so this grabs mutex, updates shared data structure, and releases mutex.
* link thread wakes up and starts running
* handler next calls `clock.set_source` and directly sets enable state
* link thread reaches the block `if (clock_link_shared_data.requested_tempo > 0)` and processes tempo change. since link has already been enabled, this is treated as a user-initiated tempo change and if there are other clients in the session, this violates the TEMPO-2 testcase.

Reply:
in other words, we need the disabled -> enabled transition to happen here on line 66, after the tempo change request has been committed.